### PR TITLE
added: hasStarted, hasEnded

### PR DESCRIPTION
**Pull request by umbri**
added: wasStart, wasEnd, useful when bench async operations

### CODE REVIEW

File: src/Ubench.php [hunk 1]
```diff
@@ -166,4 +166,12 @@ public static function readableElapsedTime($microtime, $format = null, $round =
 
         return sprintf($format, $time, $unit);
     }
+
+    public function wasEnd() {
```
Comment (devster) [inline src/Ubench.php]: Hi, and thanks for your work.
Can you update to keep the same coding style as above ?

Comment (umbri) [reply src/Ubench.php]: @devster sorry, i have fixed it in tests but forget in source, updating ...

File: src/Ubench.php [hunk 2]
```diff
@@ -166,4 +166,14 @@ public static function readableElapsedTime($microtime, $format = null, $round =
 
         return sprintf($format, $time, $unit);
     }
+
+    public function wasEnd()
```
Comment (devster) [inline src/Ubench.php]: sorry I missed that the first time but i think the correct name should be: hasEnded and hasStarted

Can you make this change?

File: tests/UbenchTest.php [hunk 3]
```diff
@@ -96,4 +98,23 @@ public function testCallableWithArguments()
         $this->assertNotNull($bench->getMemoryUsage());
         $this->assertNotNull($bench->getMemoryPeak());
     }
+
+    public function testWasStart()
+    {
+        $bench = new Ubench();
+
+        $this->assertFalse($bench->wasStart());
```
Comment (devster) [inline tests/UbenchTest.php]: You forgot the tests :) but there is no CI,  I try to setup travis really quick :)

Comment (umbri) [reply tests/UbenchTest.php]: omg, sorry, fixing ...

File: src/Ubench.php [hunk 4]
```diff
@@ -44,22 +44,33 @@ public function start()
...
-     * @param  string  $format   The format to display (printf format)
-     * @return string|float
+     * @param bool $raw
+     * @param  string $format The format to display (printf format)
+     * @return float|string
+     * @throws Exception
      */
     public function getTime($raw = false, $format = null)
     {
+        if (!$this->hasStarted())
```
Comment (devster) [inline src/Ubench.php]: Can you follow PSR2 coding style and add some brackets? even for one logic line :)

**Comment by umbri** [comment]
omg, I pushed with Exceptions throws, shell I leave or rollback ?

File: src/Ubench.php [hunk 5]
```diff
@@ -44,22 +44,33 @@ public function start()
...
+            throw new Exception("You must call start()");
```
Comment (devster) [inline src/Ubench.php]: I prefer the LogicException, that is used to informs the developper that he need to do something

**Comment by devster** [comment]
we can do this in the same PR :)

**Comment by devster** [comment]
Thanks you I'll make a new tag in fews minutes: 2.0.0

**Comment by devster** [comment]
You can already use it with the tag:  `2.0.x-dev`

**Pull request merged**